sawmill-core/src/test/java/io/logz/sawmill/processors/DeDotProcessorTest.java

```diff
@@ -0,0 +1,124 @@
+            "    \"registered\": \"Tuesday, March 8, 2016 2:35 PM\",\n" +
+            "    \"gps.latitude\": \"-83.019706\",\n" +
+            "    \"longitude\": \"6.023174\",\n" +
+            "    \"tags\": [\n" +
+            "      \"mollit\",\n" +
+            "      \"qui\",\n" +
+            "      \"anim\",\n" +
+            "      \"pariatur\",\n" +
+            "      \"pariatur\"\n" +
+            "    ],\n" +
+            "    \"range\": [\n" +
+            "      0,\n" +
+            "      1,\n" +
+            "      2,\n" +
+            "      3,\n" +
+            "      4,\n" +
+            "      5,\n" +
+            "      6,\n" +
+            "      7,\n" +
+            "      8,\n" +
+            "      9\n" +
+            "    ],\n" +
+            "    \"friends\": [{\n" +
+            "        \"id\": 0,\n" +
+            "        \"first.name\": \"Sheena James\"\n" +
+            "      },\n" +
+            "      {\n" +
+            "        \"id\": 1,\n" +
+            "        \"first.name\": \"Castaneda Hinton\"\n" +
+            "      },\n" +
+            "      {\n" +
+            "        \"id\": 2,\n" +
+            "        \"first.name\": \"Della Curry\"\n" +
+            "      }\n" +
+            "    ],\n" +
+            "    \"greeting\": \"Hello, Robin! You have 10 unread messages.\",\n" +
+            "    \"favoriteFruit\": \"banana\"\n" +
+            "  },\n" +
+            "  \"_id\": \"5c9ce370543222f1cf6b53e0\",\n" +
+            "  \"index\": 0,\n" +
+            "  \"guid\": \"d055c342-4c62-4af3-944d-541a835f79a4\",\n" +
+            "  \"isActive\": false,\n" +
+            "  \"balance\": \"$3,331.30\",\n" +
+            "  \"picture\": \"http://placehold.it/32x32\",\n" +
+            "  \"age\": 29,\n" +
+            "  \"eyeColor\": \"blue\",\n" +
+            "  \"first.name\": {\n" +
+            "    \"first\": \"Magdalena\",\n" +
+            "    \"last\": \"Stewart\"\n" +
+            "  }\n" +
+            "}";
+
+    @Test
+    public void testDefaultSeperator() throws InterruptedException {
+
+        Map<String, Object> config = new HashMap<>();
+        Doc doc = new Doc(JsonUtils.fromJsonString(Map.class,messageExample));
+        DeDotProcessor deDotProcessor = createProcessor(DeDotProcessor.class, config);
+        assertThat(deDotProcessor.process(doc).isSucceeded()).isTrue();
+
+        assertThat(doc.getSource().get("inner.object")).isNull();
+        assertThat(doc.getSource().get("first.name")).isNull();
+
+        String seperator =  "_";
+
+        assertThat(doc.getSource().get("first"+ seperator +"name")).isNotNull();
```

Comment:
lets assert the following cases:
1. keys with dots
2. nested keys with dots
3. list of maps contains keys with dots